mlflow/store/tracking/file_store.py

```diff
@@ -1208,6 +1210,51 @@ def _get_all_inputs(self, run_info: RunInfo) -> RunInputs:
 
         return RunInputs(dataset_inputs=dataset_inputs)
 
+
+    def _search_datasets(self, experiment_ids) -> List[DatasetSummary]:
+        """
+        Return all dataset summaries associated to the given experiments.
+
+        :param experiment_ids List of experiment ids to scope the search
+
+        :return A List of :py:class:`mlflow.entities.DatasetSummary` entities.
+        """
+        
+        MAX_DATASET_SUMMARIES_RESULTS = 1000
+        summary_tuples = []
+        for experiment_id in experiment_ids:
+            experiment_dir = self._get_experiment_path(experiment_id, assert_exists=True)            
+            run_dirs = list_all(
+                experiment_dir,
+                filter_func=lambda x: all(
+                    os.path.basename(os.path.normpath(x)) != reservedFolderName
+                    for reservedFolderName in FileStore.RESERVED_EXPERIMENT_FOLDERS
+                )
+                and os.path.isdir(x),
+                full_path=True,
+            )
+            for run_dir in run_dirs:
+                run_info = self._get_run_info_from_dir(run_dir)
+                run_inputs = self._get_all_inputs(run_info)
+                for dataset_input in run_inputs.dataset_inputs:
+                    dataset = dataset_input.dataset
+                    for input_tag in dataset_input.tags:
+                        if input_tag.key == MLFLOW_DATASET_CONTEXT:
+                            summary_tuples.append((experiment_id, dataset.name, dataset.digest, input_tag.value))
```

Comment:
can we use a Dataclass here instead of the tuple? IIRC Dataclasses are hashable, so can still be used with set. this will help us from having to pull out values by the indices